### PR TITLE
refactor: avoid capturing queryExit errors until the last retry attempt

### PR DESCRIPTION
**Pull request by agualis**
# Description

Goal: we want to reduce false positives in sentry issues

In the exit flow, we use a watcher to repeatedly call `queryExit` to simulate the exit transaction from the "preview modal". 
If some of this queries goes wrong (due to different causes) vue-query will retry 3 times. 
We have some random errors that happen just once but they don't in the second/third retry so they are not actually affecting the user, however we currently log this errors increasing fatal-error/transaction ratio in sentry. 

I reproduced this scenario in the generalized exit flow, where, I get random 'out-of-bounds' exceptions during queryExit simulations that do not happen after the first retry. 

Solution: In the simulation queries, instead of calling `logExitException` after the first error, we wait for the 3 vue-query retries to be attempted, so that, we will avoid this sentry issues when they don't actually affect the user flow.

It also renames `queryExit` to `simulateExit` because we had `queryExit` function in 4 layers (provider, service, handler and sdk) so it can be a bit difficult to reason about the flow and error handling for a new developer. Now `simulateExit` function shows the upper level where we handle the errors. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency changes
- [x] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

## Visual context

Please provide any relevant visual context for UI changes or additions. This could be static screenshots or a loom screencast.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have requested at least 1 review (If the PR is significant enough, use best judgement here)
- [ ] I have commented my code where relevant, particularly in hard-to-understand areas
- [ ] If package-lock.json has changes, it was intentional.
- [ ] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

File: src/providers/local/exit-pool.provider.ts [hunk 1]
```diff
@@ -385,7 +397,7 @@ export const exitPoolProvider = (
       isTxPayloadReady.value = output.txReady;
       return output;
     } catch (error) {
-      logExitException(error as Error);
+      // We will log this error (via logExitException) only after the last retry attempt (inside logOnlyAfterLastRetryAttempt)
```
Comment (garethfuller) [inline src/providers/local/exit-pool.provider.ts:400]: The problem here may be that you don't get the root error sent to Sentry, you get the error thrown below.

File: src/providers/local/exit-pool.provider.ts [hunk 2]
```diff
@@ -348,7 +360,7 @@ export const exitPoolProvider = (
   /**
    * Simulate exit transaction to get expected output and calculate price impact.
    */
-  async function queryExit() {
+  async function simulateExit() {
```
[the remark this answers is not in the document]
Comment (agualis) [reply src/providers/local/exit-pool.provider.ts:363]: Ok, I see your point.

File: src/providers/local/exit-pool.provider.ts (hunk 1 above)
Comment (agualis) [reply src/providers/local/exit-pool.provider.ts:400]: Arrggg, true. I see that vue-query also exposes `failureCount`. Probably that's the way to go.

**Comment by agualis** [comment]
Closed by #3514

**Pull request closed without merging**